backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/flightleg/BpkFlightLeg.kt

```diff
@@ -0,0 +1,266 @@
+/**
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.compose.flightleg
+
+import android.content.res.Configuration
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.BoxScope
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalLayoutDirection
+import androidx.compose.ui.semantics.contentDescription
+import androidx.compose.ui.semantics.invisibleToUser
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.text.SpanStyle
+import androidx.compose.ui.text.buildAnnotatedString
+import androidx.compose.ui.text.style.BaselineShift
+import androidx.compose.ui.text.withStyle
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.LayoutDirection
+import net.skyscanner.backpack.compose.flightleg.internal.BpkAccessoryTitleDetails
+import net.skyscanner.backpack.compose.icon.BpkIcon
+import net.skyscanner.backpack.compose.theme.BpkTheme
+import net.skyscanner.backpack.compose.tokens.Aircraft
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+
+@OptIn(ExperimentalComposeUiApi::class)
+@Composable
+fun BpkFlightLeg(
+  departureArrivalTime: String,
+  description: AnnotatedString,
+  stopsInfo: AnnotatedString,
```

Comment:
I decided to make `stopsInfo` an `AnnotatedString` as opposed to a `String` as defined in the API in Figma to make it more consistent with the other externally-stylable text in the component (description) and give more freedom for clients to style it as they need.
This also means it is easier to integrate the component when formatting texts are returned from a BFF.
For example, in Save To List feature, the BFF would return this field with styling tags (e.g.: `stopsInfo: <style0>2 stops</style0>`). 

We already have code to read these tags and return an `AnnotatedString` out of it for various style tags we support. By making this field an AnnotatedString, we can reuse that logic and reduce the use of bespoke code (this case would have to be treated as an exception to the style-tags mechanism)

Please let me know if there are any concerns

Reply:
The reason for the API proposal was to avoid exposing styling to consumers. We need to do this for the other options given the current backend responses, but ultimately this should be more an exception than a norm - this is also why there's two tables of API design with the second being the preferred option, but due to the current limitation in APIs we're not able to do this.
If we change approach this should be done across all platforms, rather than a change just for Android, so please raise it in the flightleg channel with your concerns.

Reply:
thanks! raised the discussion there